FWCore/Services/plugins/ConcurrentModuleTimer.cc

```diff
@@ -236,6 +256,9 @@ void ConcurrentModuleTimer::fillDescriptions(edm::ConfigurationDescriptions& des
   desc.addUntracked<std::vector<std::string>>("modulesToExclude", std::vector<std::string>{})
       ->setComment("Module labels to exclude from the timing measurements");
   desc.addUntracked<bool>("excludeSource", false)->setComment("Exclude the time the source is running");
+  desc.addUntracked<unsigned int>("padding", 0)->setComment("Extra possible concurrent modules beyond thread count");
```

Comment:
I'm a bit confused for what exactly the `padding` is needed? Could you elaborate some use case(s) (maybe even in the parameter comment)?

Reply:
When things went wrong, the `wait_for_all` was causing another module to be run on the thread which was already running a module. This made # concurrent modules > # threads. In order to test for that case (and keep the code from doing a buffer overflow) I added `padding`.

Reply:
Extended the comments